libs/execution_context/ebpf_native.c

```diff
@@ -201,8 +201,12 @@ void
 ebpf_native_acquire_reference(_Inout_ ebpf_native_module_t* module)
 {
     ebpf_assert(module->base.marker == _ebpf_native_marker);
+
+#pragma warning(push)
+#pragma warning(disable : 4189)
```

Comment:
Can you add a comment what is the warning about.

Reply:
Suggestion: Alternatively, we can add `UNREFERENCED_PARAMETER(new_ref_count)` instead of suppressing the warning.

Or change it to `ebpf_assert(ebpf_interlocked_increment_int32(&module->base.reference_count) != 1);`

Reply:
> Suggestion: Alternatively, we can add `UNREFERENCED_PARAMETER(new_ref_count)` instead of suppressing the warning.

This is valid for function parameters, not local variables, correct?

> Or change it to `ebpf_assert(ebpf_interlocked_increment_int32(&module->base.reference_count) != 1);`

This was one of the first obvious (of the many) attempts in suppressing the "unreferenced" compiler warning (then treated as an error) but did not work. E.g., with this implementation, for the cmake-release build you'd get:

```
D:\a\ebpf-for-windows\ebpf-for-windows\libs\execution_context\ebpf_native.c(201,61): error C2220: the following warning is treated as an error [D:\a\ebpf-for-windows\ebpf-for-windows\build\libs\execution_context\execution_context_kernel.vcxproj]
D:\a\ebpf-for-windows\ebpf-for-windows\libs\execution_context\ebpf_native.c(201,61): warning C4100: 'module': unreferenced formal parameter [D:\a\ebpf-for-windows\ebpf-for-windows\build\libs\execution_context\execution_context_kernel.vcxproj]
```
then, you'd need to suppress C4100.

Since the warning appears in Release mode only and is suppressed in a very controlled way without affecting any runtime functionality, this approach shouldn't be an issue.

Reply:
After some further tests, it looks like the `UNREFERENCED_PARAMETER()` macro *would* work (i.e. suppressing the "unreferenced" warning), even though it's theoretically designed for function parameters.
After an internal sync, we decided to keep the actual solution, rather than using the aforementioned macro in a non-explicitly documented way (ideally the xDKs would have had an exact copy, just renamed as `UNREFERENCED_VARIABLE()`).